core/src/main/scala/kafka/log/LogValidator.scala

```diff
@@ -102,36 +111,48 @@ private[kafka] object LogValidator {
 
   private def assignOffsetsNonCompressed(records: MemoryRecords,
                                          offsetCounter: LongRef,
-                                         now: Long,
+                                         currentTimestamp: Long,
                                          compactedTopic: Boolean,
                                          timestampType: TimestampType,
                                          timestampDiffMaxMs: Long): ValidationAndOffsetAssignResult = {
-    var maxTimestamp = Record.NO_TIMESTAMP
+    var maxTimestamp = RecordBatch.NO_TIMESTAMP
     var offsetOfMaxTimestamp = -1L
-    val firstOffset = offsetCounter.value
+    val initialOffset = offsetCounter.value
 
-    for (entry <- records.shallowEntries.asScala) {
-      val record = entry.record
-      validateKey(record, compactedTopic)
+    for (batch <- records.batches.asScala) {
+      ensureNonTransactional(batch)
```

Comment:
This was intended to be temporary until we have the transactional portion of KIP-98. I didn't want to get too far into that validation in this patch if possible. Does that seem reasonable?